src/AppInstallerCommonCore/MSStoreDownload.cpp

```diff
@@ -0,0 +1,491 @@
+                for (const auto& arch : architectures)
+                {
+                    catalogPackage.Architectures.emplace_back(Utility::ConvertToArchitectureEnum(arch));
+                }
+
+                // Languages
+                auto languages = JSON::GetRawStringArrayFromJsonNode(packageEntry, JSON::GetUtilityString(Languages));
+                for (const auto& language : languages)
+                {
+                    catalogPackage.Languages.emplace_back(language);
+                }
+
+                // Package Format
+                std::optional<std::string> packageFormat = JSON::GetRawStringValueFromJsonNode(packageEntry, JSON::GetUtilityString(PackageFormat));
+                if (packageFormat)
+                {
+                    catalogPackage.PackageFormat = ConvertToPackageFormatEnum(packageFormat.value());
+                }
+
+                // Content Id
+                std::optional<std::string> contentId = JSON::GetRawStringValueFromJsonNode(packageEntry, JSON::GetUtilityString(ContentId));
+                if (contentId)
+                {
+                    catalogPackage.ContentId = contentId.value();
+                }
+
+                // WuCategoryId
+                std::optional<std::reference_wrapper<const web::json::value>> fulfillmentData = JSON::GetJsonValueFromNode(propertiesValue, JSON::GetUtilityString(FulfillmentData));
+                if (fulfillmentData)
+                {
+                    const auto& fulfillmentDataValue = fulfillmentData.value().get();
+                    std::optional<std::string> wuCategoryId = JSON::GetRawStringValueFromJsonNode(fulfillmentDataValue, JSON::GetUtilityString(WuCategoryId));
+                    if (JSON::IsValidNonEmptyStringValue(wuCategoryId))
+                    {
+                        catalogPackage.WuCategoryId = wuCategoryId.value();
+                    }
+                }
+
+                displayCatalogPackages.emplace_back(catalogPackage);
+            }
+
+            return displayCatalogPackages;
+        }
+        catch (const std::exception& e)
+        {
+            AICLI_LOG(Core, Error, << "Error encountered while deserializing MSStore request. Reason: " << e.what());
+        }
+        catch (...)
+        {
+            AICLI_LOG(Core, Error, << "Received invalid information.");
+        }
+    }
+
+    std::string GetMSStoreCatalogRestApi(const std::string& productId, const std::string& locale)
+    {
+        std::string languageValue = !locale.empty() ? locale : std::string{ Neutral };
+        std::string market = AppInstaller::Runtime::GetOSRegion();
+
+        auto restEndpoint = AppInstaller::Utility::Format(std::string{ MSStoreCatalogRestApi }, productId, Details, market, languageValue);
+        THROW_HR_IF(APPINSTALLER_CLI_ERROR_RESTSOURCE_INVALID_URL, !AppInstaller::Rest::IsValidUri(JSON::GetUtilityString(restEndpoint)));

```

Comment:
>THROW_HR_IF(APPINSTALLER_CLI_ERROR_RESTSOURCE_INVALID_URL, !AppInstaller::Rest::IsValidUri(JSON::GetUtilityString(restEndpoint))); [](http://example.com/codeflow?start=8&length=130)

Not needed since we are the one generating the uri